reconcile/test/test_ocm_clusters.py

```diff
@@ -0,0 +1,224 @@
+            rs,
+            {
+                'cluster1': {
+                    'spec': self.clusters[0]['spec'],
+                    'network': self.clusters[0]['network'],
+                    'consoleUrl': '',
+                    'serverUrl': '',
+                    'elbFQDN': '',
+                    'prometheusUrl': '',
+                    'alertmanagerUrl': ''
+                }
+            }
+        )
+
+
+class TestGetClusterUpdateSpec(TestCase):
+    def setUp(self):
+        self.clusters = [
+            fxt.get_anymarkup('cluster1.yml')
+        ]
+
+    def test_no_changes(self):
+        self.assertEqual(
+            occ.get_cluster_update_spec(
+                'cluster1',
+                self.clusters[0],
+                self.clusters[0]
+            ),
+            ({}, False)
+        )
+
+    def test_valid_change(self):
+        desired = deepcopy(self.clusters[0])
+        desired['spec']['instance_type'] = 'm42.superlarge'
+        print(desired)
+        print(self.clusters[0])
+        self.assertEqual(
+            occ.get_cluster_update_spec(
+                'cluster1',
+                self.clusters[0],
+                desired,
+            ),
+            ({'instance_type': 'm42.superlarge'}, False)
+        )
+
+    def test_changed_network(self):
+        desired = deepcopy(self.clusters[0])
+        self.clusters[0]['network']['vpc'] = '10.0.0.0/8'
+        self.assertEqual(
+            occ.get_cluster_update_spec(
+                'cluster1', self.clusters[0], desired
+            ),
+            ({}, True)
+        )
+
+    def test_changed_spec_bad(self):
+        desired = deepcopy(self.clusters[0])
+        desired['spec']['multi_az'] = not desired['spec']['multi_az']
+        self.assertTrue(
+            occ.get_cluster_update_spec(
+                'cluster1', self.clusters[0], desired
+            )[1],
+        )
+
+
+class TestRun(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.clusters = [fxt.get_anymarkup('cluster1.yml')]
+        self.clusters[0]['ocm']['name'] = 'ocm-nonexisting'
+        self.clusters[0]['path'] = '/openshift/mycluster/cluster.yml'
+        self.mock_callable(
+            queries, 'get_app_interface_settings'
+        ).for_call().to_return_value({}).and_assert_called_once()
+        self.get_clusters = self.mock_callable(
+            queries, 'get_clusters'
+        ).for_call().to_return_value(self.clusters).and_assert_called_once()
+        self.ocmmap = StrictMock(ocmmod.OCMMap)
+        self.ocm = StrictMock(ocmmod.OCM)
+        self.mock_constructor(ocmmod, 'OCMMap').to_return_value(self.ocmmap)
+        self.mock_callable(
+            self.ocmmap, 'get'
+        ).for_call('cluster1').to_return_value(self.ocm)
+        self.update_cluster = self.mock_callable(
+            self.ocm, 'update_cluster'
+        ).to_return_value(None)
+        self.mock_callable(
+            sys, 'exit'
+        ).to_raise(ValueError)
```

Comment:
This is because sys.exit will exit the test execution? The only thing I'm wondering is if it would make sense to raise some less common exception in it's place to avoid cases where a `ValueError` could really be legitimately raise? Definitely a solid case for avoiding `sys.exit` in the future as we've discussed.